### PR TITLE
Rename VPC security group types

### LINKED ISSUE #1770: Reason to keep VPC and ResourceGroup param in VPC Security Group
/kind feature
/area provider/ibmcloud

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]
This issue is to continue the discussion on the question asked [here](https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/pull/1752#issuecomment-2090543765)

> @cjschaef I just have a comment on [VPC](https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/732b7732e3aa883d96e88eb03fd932d54030bace/api/v1beta2/types.go#L296) and [ResourceGroup](https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/732b7732e3aa883d96e88eb03fd932d54030bace/api/v1beta2/types.go#L284) properties of type SecurityGroup. are you planning to pass a different VPC and Resource Group than the one cluster is using? Just wanted to understand what is the intention for the properties?

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by dharaneeshvrd**
Remove VPC and resource group field from VPC security group since it is causing by providing an option to pass a different resource than the one used in cluster

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1770 

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
Rename VPC security group types and remove VPC and resource group field from VPC security group since it is causing confusion by providing an option to pass a different resource than the one used in cluster
 
```

### CODE REVIEW

File: api/v1beta2/types.go [hunk 1]
```diff
@@ -268,24 +268,16 @@ type PortRange struct {
...
 // +kubebuilder:validation:XValidation:rule="has(self.id) || has(self.name)",message="either an id or name must be specified"
-type SecurityGroup struct {
+type VPCSecurityGroup struct {
 	// id of the Security Group.
 	// +optional
 	ID *string `json:"id,omitempty"`
 
-	// name of the Security Group.
-	// +optional
-	Name *string `json:"name,omitempty"`
```
Comment (Amulyam24) [inline api/v1beta2/types.go:280]: @dharaneeshvrd, is this by mistake? you intended to delete VPC and ResouceGroup right

Comment (dharaneeshvrd) [reply api/v1beta2/types.go:280]: Yes, corrected, thanks @Amulyam24 !

**Review by Amulyam24 — APPROVED** [review]
/lgtm

Please fill in the release note.

File: api/v1beta2/types.go [hunk 2]
```diff
@@ -190,62 +190,62 @@ var (
...
-// SecurityGroupRuleAction represents the actions for a Security Group Rule.
+// VPCSecurityGroupRuleAction represents the actions for a Security Group Rule.
 // +kubebuilder:validation:Enum=allow;deny
-type SecurityGroupRuleAction string
+type VPCSecurityGroupRuleAction string
 
 const (
 	// SecurityGroupRuleActionAllow defines that the Rule should allow traffic.
-	SecurityGroupRuleActionAllow SecurityGroupRuleAction = vpcv1.NetworkACLRuleActionAllowConst
+	SecurityGroupRuleActionAllow VPCSecurityGroupRuleAction = vpcv1.NetworkACLRuleActionAllowConst
```
Comment (mkumatag) [inline api/v1beta2/types.go:199]: Do you wanna rename the variable names as well to keep them also in the same format?

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: api/v1beta2/types.go (hunk 2 above)
Comment (dharaneeshvrd) [reply api/v1beta2/types.go]: Yes renamed all of them, thanks!

**Pull request merged**